File: src/main/java/gg/mineral/practice/scoreboard/impl/InMatchScoreboard.java
package gg.mineral.practice.scoreboard.impl;

import gg.mineral.practice.entity.Profile;
import gg.mineral.practice.scoreboard.Scoreboard;
import gg.mineral.practice.scoreboard.ScoreboardHandler;
import gg.mineral.practice.util.messages.CC;
import org.bukkit.craftbukkit.v1_8_R3.entity.CraftPlayer;
import org.jetbrains.annotations.NotNull;

public class InMatchScoreboard
        implements Scoreboard {

    public static final Scoreboard INSTANCE = new InMatchScoreboard();

    @Override
    public void updateBoard(ScoreboardHandler board, @NotNull Profile profile) {
        board.updateTitle(CC.PRIMARY + CC.B + "Mineral");
        board.updateLines(CC.BOARD_SEPARATOR,
                CC.ACCENT + "Opponent: " + CC.SECONDARY + profile.getMatch().getOpponent(profile).getName(),
                CC.ACCENT + "Your Ping: " + CC.SECONDARY + ((CraftPlayer) board.getPlayer()).getHandle().ping,
                CC.ACCENT + "Their Ping: " + CC.SECONDARY
                        + profile.getMatch().getOpponent(profile).getPlayer().getHandle().ping,
                CC.SPACER,
                CC.SECONDARY + "mineral.gg",
                CC.BOARD_SEPARATOR);
    }
}

File: src/main/kotlin/gg/mineral/practice/scoreboard/impl/InMatchScoreboard.kt
package gg.mineral.practice.scoreboard.impl

import gg.mineral.practice.entity.Profile
import gg.mineral.practice.managers.ProfileManager.getPing
import gg.mineral.practice.scoreboard.Scoreboard
import gg.mineral.practice.scoreboard.ScoreboardHandler
import gg.mineral.practice.util.messages.CC

class InMatchScoreboard : Scoreboard {
    override fun updateBoard(board: ScoreboardHandler, profile: Profile) {
        board.updateTitle(CC.PRIMARY + CC.B + "Mineral")
        board.updateLines(
            CC.BOARD_SEPARATOR,
            CC.ACCENT + "Opponent: " + CC.SECONDARY + profile.match!!.getOpponent(profile)?.name,
            CC.ACCENT + "Your Ping: " + CC.SECONDARY + board.player.getPing(),
            (CC.ACCENT + "Their Ping: " + CC.SECONDARY
                    + profile.match?.getOpponent(profile)?.player?.getPing()),
            CC.SPACER,
            CC.SECONDARY + "mineral.gg",
            CC.BOARD_SEPARATOR
        )
    }

    companion object {
        val INSTANCE: Scoreboard = InMatchScoreboard()
    }
}
